packages/router/router.ts

```diff
@@ -2450,15 +2450,27 @@ export function createRouter(init: RouterInit): Router {
     };
   }
 
+  function getScrollKey(location: Location, matches: AgnosticDataRouteMatch[]) {
+    if (getScrollRestorationKey) {
+      let loc = {
+        ...location,
+        pathname:
+          stripBasename(location.pathname, basename) || location.pathname,
```

Comment:
Note to self: see about moving this to the RR layer since that's where the `basename` stripping should happen to stay consistent